### ISSUE TITLE
False positive from JavadocType: Unused @param tag

### ISSUE THREAD

**Issue by turbanoff** [checkstyle/checkstyle#17332]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadoctype.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac WithNative.java

$ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="authorFormat" value="\S"/>
            <property name="allowMissingParamTags" value="true"/>
            <property name="allowUnknownTags" value="true"/>
        </module>
    </module>
</module>

$ cat WithNative.java
import java.util.Collection;

public class WithNative {

    /**
     * Creates Plugin
     *
     * @param chartComponent chart component
     * @return plugin
     */
    private native String createPlugin(Long chartComponent) /*-{
        return pluginModel;
    }-*/;

    private interface PluginModel {
    }
}

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-10.26.1-all.jar -c config.xml WithNative.java
Starting audit...
[ERROR] D:\temp\checkstyle\WithNative.java:8:8: Unused @param tag for 'chartComponent'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**
Method `createPlugin` has proper javadoc. I expect no errors.

**Comment by turbanoff** [comment]
It worked fine in 10.20.0. It's a regression.

**Comment by romani** [comment]
@turbanoff , do you have any open source project with native methods and good javadocs?
To let us add it to our testing.and make sure we will fix it without any other regression.

**Comment by turbanoff** [comment]
OpenJDK have good javadoc.
https://github.com/openjdk/jdk/blob/a201be8555c57f07b86f470df4699e1b9dd6bd3c/src/java.base/share/classes/java/lang/Class.java#L756

**Comment by romani** [comment]
Strange, we already have open jdk in our regression testing,and we did not catch it. 
Ok, we will see during PR, diff on openjdk.

**Comment by turbanoff** [comment]
Looks like there should be some specific pattern to trigger the issue.
GWT-style native method
```
    private native String createPlugin(Long chartComponent) /*-{
        return pluginModel;
    }-*/;
```
And nested class declared after that.
I tried to search for open-source GWT projects, but wasn't able to find active ones, which use such native methods.

**Comment by romani** [comment]
Whole logic is at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java , everyone is welcome to send PR.

**Comment by Atharv3221** [comment]
I am working on it

**Comment by Atharv3221** [comment]
There are bugs 

```
import java.util.Collection;

public class WithNative {

    /**
     * Creates Plugin
     *
     * @param chartComponent chart component
     * @return plugin
     */
    private native String createPlugin(Long chartComponent) /*-{
        return pluginModel;
    }-*/;

    private interface PluginModel {
    }
}
```

bug 1: we never actually check for PARAMETERS, we only check for TYPE_PARAMETER
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java#L337
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java#L353

bug 2:    
```
private native String createPlugin(Long chartComponent) /*-{
        return pluginModel;
    }-*/;
```
This whole is treated as comment block

**Comment by Atharv3221** [comment]
@romani 
False positive is fixed

Was this Check developed  only to check TYPE_PARAMETERS or there should be also be checking for PARAMETER ??

**Comment by romani** [comment]
Check is developed to do violations, what tokens it operates inside, nobody care.

**Comment by nrmancuso** [comment]
Fixed via #17489
